lib/compress.js

```diff
@@ -176,6 +176,21 @@ merge(Compressor.prototype, {
         }
     };
 
+    // we shouldn't compress (1,func)(something) to
+    // func(something) because that changes the meaning of
+    // the func (becomes lexical instead of global).
+    function check_function_call(parent, orig, val) {
```

Comment:
Both the words "lexical" and "scope" have little to do with what's going on here. It's about the kind of Reference returned by the expression.